api/download.go

```diff
@@ -14,123 +13,94 @@ import (
 const maxIPsPerDay = 50
 
 // DownloadURL returns a signed URL to download a purchased asset.
-func (a *API) DownloadURL(ctx context.Context, w http.ResponseWriter, r *http.Request) {
-	id := kami.Param(ctx, "id")
-	log := gcontext.GetLogger(ctx).WithField("download_id", id)
+func (a *API) DownloadURL(w http.ResponseWriter, r *http.Request) error {
+	ctx := r.Context()
+	downloadID := chi.URLParam(r, "download_id")
+	logEntrySetField(r, "download_id", downloadID)
 	claims := gcontext.GetClaims(ctx)
 	assets := gcontext.GetAssetStore(ctx)
 
 	download := &models.Download{}
-	if result := a.db.Where("id = ?", id).First(download); result.Error != nil {
+	if result := a.db.Where("id = ?", downloadID).First(download); result.Error != nil {
 		if result.RecordNotFound() {
-			log.Debug("Requested record that doesn't exist")
-			notFoundError(w, "Download not found")
-		} else {
-			log.WithError(result.Error).Warnf("Error while querying database: %s", result.Error.Error())
-			internalServerError(w, "Error during database query: %v", result.Error)
+			return notFoundError("Download not found")
 		}
-		return
+		return internalServerError("Error during database query: %v", result.Error)
```

Comment:
this would leak the result.Error right? We'd want to use the `Internal` methods you have.

Same comment for the other similar methods